.github/workflows/ci.yml

```diff
@@ -0,0 +1,141 @@
+name: ci
+on:
+  push:
+    # branches: [main]
+    branches: 
+     - "**"       # Run on any pushed branch for testing
+    tags:
+      - "v*"       # Run on any pushed tag
+  pull_request:
+    branches: [main]
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.24.2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v7
+        with:
+          version: v2.0
+          args: --timeout 5m --verbose
+        
+  build:
+    needs: format
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-go@v5
+      with:
+        go-version: 1.24.2
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Build and export Docker image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        tags: ${{ github.repository }}:latest
+        outputs: type=docker,dest=/tmp/image.tar
+    - name: Upload Docker image artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: image
+        path: /tmp/image.tar
+
+  run_tests:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        token: ["", "token"]
```

Comment:
let's simplify it like here - https://github.com/reductstore/reduct-rs/blob/main/.github/workflows/ci.yml#L33

we run tests for the main and latest  versions with the API token and license key.